### ISSUE TITLE
[CI] Replace Ubuntu 18.04 with 16.04

### ISSUE THREAD

**Issue by flibitijibibo** [TerryCavanagh/VVVVVV#67]
This one's for @ePirat, @charliebruce, and also @cybik since this will pretty much directly affect our [ANGLE-Builds](https://github.com/cybik/angle-bootstraps) system for FNA and friends.

Basically, 16.04 is supposed to be supported by GitHub Actions, but for some reason it doesn't work. VVVVVV needs it for compatibility testing, ANGLE needs it so our builds target a low enough glibc/libstdc++ for people to safely distribute with, so we need to find out why 16.04 is not cooperating with us.

**Comment by cybik** [comment]
https://github.com/actions/virtual-environments for reference

**Comment by flibitijibibo** [comment]
Turns out the bug was that ninja was too old, we have ANGLE building with 16.04 now. I think our CMakeLists needs to set some policies to deal with other things, but cybik's working on it.

**Comment by cybik** [comment]
Issue diagnosed: SDL too old.

https://github.com/cybik/VVVVVV/commit/97aa4a217b2d6191ac1a9ff5221672de9b64fbcc/checks?check_suite_id=398221029

**Comment by cybik** [comment]
At a glance, it doesn't seem like there's a PPA for libsdl2.

Unfortunately, this may mean I can do my shenanigans and quick-compile an SDL2 version inside the CI expressly for this. My apologies.

**Comment by AB036** [comment]
I managed to compile the game on Ubuntu 16.04 (xenial). I had to compile libsdl2 and libsdl2-mixer from sources because of SDL_RenderSetIntegerScale (I used the latest versions).

The sound wasn't working at first because I didn't have the required libraries. The configure script of libsdl2-mixer should tell you what formats it supports. Make sure that ogg and wav are enabled.

Ogg support wasn't enabled, but it worked once I installed libvorbis-dev with apt.
